### PR TITLE
FEAT: Added Authentication

### PR DESCRIPTION
**Pull request by kotesh-arya**
## Detailed features of this PR :
-   Integrated Authentication for User-specific Protected routes (Cart, Wishlist)

### CODE REVIEW

File: src/pages/SignUp/SignUp.jsx [hunk 1]
```diff
@@ -1,19 +1,61 @@
...
 import { NavBar } from "../HomePage/HomePageComponents";
 import { Link } from "react-router-dom";
 import SignUpCSS from "../SignUp/SignUp.module.css";
+import { useAuth } from "../../contexts/authContext";
 function SignUp() {
+  const { signupUser } = useAuth();
+  // const [email, setEmail] = useState("");
+  // const [password, setPassword] = useState("");
+  // const [firstName, setFirstName] = useState("");
+  // const [lastName, setLastName] = useState("");
```
Comment (MaruthiPothuganti) [inline src/pages/SignUp/SignUp.jsx]: Unwanted Commented code can be removed

**Review by MaruthiPothuganti — APPROVED** [review]
Code looks good to merge.

File: src/App.js [hunk 2]
```diff
@@ -17,10 +18,27 @@ function App() {
...
-        <Route path="/WishList" element={<WishList />}>
+        <Route path="/UserInfo" element={<UserInfo />}>
+          UserInfo
+        </Route>
+        <Route
+          path="/WishList"
+          element={
+            <RequiresAuth>
+              <WishList />
+            </RequiresAuth>
```
Comment (ivinayakg) [inline src/App.js:31]: you can create nested routes where you need to run the RequiresAuth only once,
read about auth with outlet from react-router
For now its fine

File: src/index.js [hunk 3]
```diff
@@ -14,11 +15,13 @@ ReactDOM.render(
...
       <CartProvider>
-        <WishlistProvider>
-          <FilterProvider>
-            <App />
-          </FilterProvider>
-        </WishlistProvider>
+        <AuthProvider>
+          <WishlistProvider>
+            <FilterProvider>
+              <App />
```
Comment (ivinayakg) [inline src/index.js]: for now its fine, but in future try using one or at most 2 contexts, that will look better

File: src/pages/Cart/Cart.module.css [hunk 4]
```diff
@@ -1,155 +1,148 @@
...
 .horizontal-card {
-    margin: 2rem 0 0 0;
-    width: 38rem;
+  margin: 2rem 0 0 0;
+  width: 38rem;
 }
 
 .cart-item-title {
-    color: var(--primary-color);
+  color: var(--primary-color);
```
Comment (ivinayakg) [inline src/pages/Cart/Cart.module.css:29]: great that you are doing themeing

File: src/pages/SignIn/SignIn.jsx [hunk 5]
```diff
@@ -1,55 +1,105 @@
...
+                setFormData((prev) => ({ ...prev, email: e.target.value }))
+              }
+            />
+          </div>
+          <div className={SignInCSS["input"]}>
+            <label htmlFor="input-box">Password</label>
+            <input
+              className={`input-box basic ${SignInCSS["text-field"]}`}
+              type="password"
+              placeholder="************"
```
Comment (ivinayakg) [inline src/pages/SignIn/SignIn.jsx]: in the placeholder type password, as when a person with disability uses it it will say this is the field for password

**Review by ivinayakg — APPROVED** [review]
your code is good, i gave some suggestions for improvement
For its good with this only also, but in future you can try and look into those. 
Great job 👍

File: src/App.js (hunk 2 above)
Comment (kotesh-arya) [reply src/App.js:31]: Sure. I will look into that 👍🏻

File: src/index.js (hunk 3 above)
Comment (kotesh-arya) [reply src/index.js]: Sure, I will


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Tags:** Efficient

**Pull request merged**